cmd/applyconfig/applyconfig.go

```diff
@@ -119,13 +153,23 @@ type configApplier struct {
 	context    string
 	path       string
 	user       string
-	dry        bool
+	dry        dryRunMethod
 }
 
-func makeOcApply(kubeConfig, context, path, user string, dry bool) *exec.Cmd {
+func makeOcApply(kubeConfig, context, path, user string, dry dryRunMethod) *exec.Cmd {
 	cmd := makeOcCommand(ocApply, kubeConfig, context, path, user)
-	if dry {
-		cmd.Args = append(cmd.Args, "--dry-run")
+	switch dry {
+	case dryAuto:
+		logrus.Warn("BUG: Automated dryrun detection should be performed earlier; Using server-side validation")
+		fallthrough
+	case dryServer:
+		cmd.Args = append(cmd.Args, "--dry-run=server", "--validate=true")
```

Comment:
Some openshift resources still do not have proper openapi schema, we can not use validate=true (I can tell from experience, because I usually use `kubectl` and it reguarly needs a `--validate=false` (it defaults to true)`

Most prominent example are the various openshift api groups that can be accessed by using no api group rather than the real one

Reply:
Good insight. I only really tested it on our app.ci manifests, and on a simple eyeballing the results did not look that bad, but I haven't looked at the details too much.

Could you point me to one of our manifests where this would be currently unfixable?

Reply:
It is fixable, it's just quite a bit of a chore :)

Reply:
(I think, I am not entirely sure tbh. Try it out :) Historically, it was just broken and until today, `oc` defaults the `--validate` flag to `false`. )